File: WorkoutApp/Models/Record.cs
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.Bson;

namespace WorkoutApp.Models
{
    public class Record
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? id { get; set; }
        public string workoutId { get; set; } = null!;
        public string dayId { get; set; } = null!;
        public int? setIndex { get; set; } = null;
        public int? repIndex { get; set; } = null;
        public string excerciseName { get; set; } = null!;
        public int? numReps { get; set; } = null;
        public int? weight { get; set; } = null;
        public DateTime dateCompleted { get; set; }
    }
}

File: WorkoutApp/ClientApp/src/app/classes/record.ts
export class Record {
  id?: string;
  dayId?: string;
  workoutId?: string;
  setIndex?: number;
  repIndex?: number;
  excerciseName?: string;
  numReps?: number;
  weight?: number;
  dateCompleted?: Date;

  constructor() {
    this.dayId = "";
    this.workoutId = "";
    this.excerciseName = "";
    this.numReps = 0;
    this.weight = 0;
  }
}
